components/bl-marquee-component/src/index.js

```diff
@@ -0,0 +1,99 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const DEFAULT_COUNT = 'infinite';
+const PlayState = { RUN: 'running', PAUSE: 'paused' };
+
+export default function MarqueeComponent({ component, elRef, eventHandlers, pods }) {
+  const {
+    classList, style, display, start, pauseOnHover, pauseOnClick, direction, speed, delay, loop, gradient,
+  } = component;
+  const { onFinish, onCycleComplete } = eventHandlers;
+
+  const children = pods['marqueeContent'].render();
+
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [marqueeWidth, setMarqueeWidth] = useState(0);
+  const [hasFinished, setHasFinished] = useState(false);
+  const [play, setPlay] = useState(start);
+
+  const marqueeRef = useRef(null);
+
+  useEffect(() => {
+    if (marqueeRef.current && elRef.current) {
+      const calculateWidth = () => {
+        setContainerWidth(elRef.current.getBoundingClientRect().width);
+        setMarqueeWidth(marqueeRef.current.getBoundingClientRect().width);
```

Comment:
If he hide and show again component using `display` property, this code still works well?